src/test/run-make/sgx-process-termination/accept.rs

```diff
@@ -0,0 +1,15 @@
+// program should terminate even if a thread is blocked on I/O.
+// https://github.com/fortanix/rust-sgx/issues/109
+
+use std::{net::TcpListener, sync::mpsc, thread};
+
+fn main() {
+    let (tx, rx) = mpsc::channel();
+    thread::spawn(move || {
+        let listen = TcpListener::bind("0:0").unwrap();
+        tx.send(()).unwrap();
+        while let Ok(_) = listen.accept() {}
```

Comment:
True, but I think there is no reasonable way to avoid that race: we need one thread to be blocked on some I/O and another thread to cause process exit. While the first thread is blocked on I/O it cannot notify the other thread (by definition), and the second thread does not have a way of knowing the first thread is truly blocked on I/O on its own unless we are willing to check `/proc/self/task/*/stat` or something similar.

I think we can reasonably expect that the main thread does not exit before the spawned thread blocks on `accept()` in most cases.

Reply:
Maybe you can do `thread::yield_now` a bunch after `recv`

Reply:
I think the `println!` is sufficient, but `thread::yield_now` maybe more readable.